src/hssm/rl/likelihoods/builder.py

```diff
@@ -0,0 +1,548 @@
+
+        # Apply pre-vmapped function to compute parameter values
+        computed_values = vmapped_func(subj_trials)
+
+        # Reshape back to 2D (total_trials, 1) for concatenation
+        return computed_values.reshape((-1, 1))
+
+    def logp(data, *args) -> np.ndarray:
+        """Compute the log-likelihood for the RLDM model for each trial.
+
+        This function computes the full log-likelihood for a reinforcement learning
+        drift-diffusion model (RLDM). It first computes any required intermediate
+        parameters (such as drift rates) for each trial, and then evaluates the
+        sequential sampling model (SSM) likelihood using these parameters.
+
+        Parameters
+        ----------
+        data : np.ndarray
+            A 2D array containing trial data.
+        *args :
+            Model parameters included in list_params and extra_fields.
+
+        Returns
+        -------
+        np.ndarray
+            The computed log-likelihoods for each trial, reshaped as a 2D array.
+        """
+        # Get column indices for SSM logp function
+        # Identifies computed vs available params
+        ssm_logp_func_colidxs = _get_column_indices_with_computed(
+            ssm_logp_func,
+            data_cols,
+            list_params,
+            extra_fields,
+        )
+
+        # Validate that all computed parameters have compute functions
+        _validate_computed_parameters(ssm_logp_func, ssm_logp_func_colidxs.computed)
+
+        computed_param_values = (
+            {
+                param_name: compute_parameter(param_name, data, args)
+                for param_name in ssm_logp_func_colidxs.computed
+            }
+            if hasattr(ssm_logp_func, "computed") and ssm_logp_func.computed
+            else {}
+        )
+
+        # Extract non-computed parameters
+        non_computed_args = _collect_cols_arrays(
+            data, args, ssm_logp_func_colidxs.colidxs
+        )
+
+        # Build final parameter matrix maintaining order from ssm_logp_func.inputs
+        # This ensures parameters appear in the sequence expected by the SSM likelihood
+        def get_param_array(param_name: str) -> jnp.ndarray:
+            """Get parameter array from computed values or non-computed args."""
+            if param_name in computed_param_values:
+                return computed_param_values[param_name]
+            # Use non-computed value (from data or args)
+            # Find its position in non_computed_args based on colidxs order
+            colidxs_keys = list(ssm_logp_func_colidxs.colidxs.keys())
+            idx = colidxs_keys.index(param_name)
+            return non_computed_args[idx].reshape((-1, 1))
+
+        param_arrays = [get_param_array(name) for name in ssm_logp_func.inputs]
+
+        # Stack all parameters into final matrix
+        lan_matrix = jnp.concatenate(param_arrays, axis=1)
+        return ssm_logp_func(lan_matrix)
```

Comment:
ok just one structural question here: 

This assumes the inputs to the `ssm_logp_fun` as a matrix, so implies that `ssm_logp_fun` is a LAN (or has that structure). 

We should make sure this also works with our `blackbox` and `analytical` likelihoods, which may demand another version of the outer function here.